cfgov/v1/management/commands/archive_pages.py

```diff
@@ -0,0 +1,124 @@
+from django.core.management.base import BaseCommand, CommandError
+from django.db import transaction
+from django.db.models import Q
+from django.http import Http404
+from django.utils import timezone
+
+from wagtail.core.models import Site
+
+from dateutil.relativedelta import relativedelta
+
+
+def path_without_leading_trailing_slashes(path):
+    return path.lstrip("/").rstrip("/")
+
+
+class Command(BaseCommand):
+    help = "Mark archivable pages within a filterable list as archived"
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "url_path",
+            nargs="+",
+            type=path_without_leading_trailing_slashes,
+            help=(
+                "The URL path (without leading /) of a filterable list page "
+                "whose content should be archived if it was published the "
+                "specified number of years, months, and days ago. "
+                "The filerable list page at the URL will not be archived."
+            ),
+        )
+        parser.add_argument(
+            "--years",
+            type=int,
+            default=2,
+            help=(
+                "Archive content published this number of years ago. "
+                "Combines with --months --days. Default: 2"
+            ),
+        )
+        parser.add_argument(
+            "--months",
+            type=int,
+            default=0,
+            help=(
+                "Archive content published this number of months ago. "
+                "Combines with --years --days. Default: 0"
+            ),
+        )
+        parser.add_argument(
+            "--days",
+            type=int,
+            default=0,
+            help=(
+                "Archive content published this number of days ago. "
+                "Combines with --years --months. Default: 0"
+            ),
+        )
+        parser.add_argument(
+            "--by-published-date",
+            choices=["first", "last"],
+            default="first",
+            help=(
+                "Archive based on either first or last published date. "
+                "Default: first"
+            ),
+        )
+
+    def handle(self, *args, **options):
+        url_paths = options["url_path"]
+
+        # Get the current date/time and then get our cutoff date for archiving
+        # based on it.
+        archived_at = timezone.now()
+        cutoff_date = archived_at - relativedelta(
+            years=options["years"],
+            months=options["months"],
+            days=options["days"],
+        )
+
+        # Construct a Q object to filter on based on this command-line argument
```

Comment:
Trivial: consistency of ending complete sentence comments with a period.

```suggestion
        # Construct a Q object to filter on based on this command-line argument.
```

Reply:
Done in 1e70b3164.